Tasks/Maven/maven2.js

```diff
@@ -133,6 +132,34 @@ function getEndpointDetails(inputFieldName) {
     };
 }
 
+// The endpoint stores the auth details as JSON. Unfortunately the structure of the JSON has changed through time, namely the keys were sometimes Upercase
+// To work around this, we can perform case insenstive checks in the property dictionary of the object. Note that the PowerShell implementation does not suffer from this problem.
+function getAuthParameter(endpoint, paramName) {
+
+    var paramValue = null;
+    var auth = tl.getEndpointAuthorization(endpoint, false);
+
+    if (auth.scheme != "UsernamePassword") {
+        throw new Error("The authorization scheme " + auth.scheme + " is not supported for a SonarQube endpoint. Please use a username and a password.");
+    }
+
+    var parameters = Object.getOwnPropertyNames(auth['parameters']);
+
+    var keyName;
+    parameters.some(function (key) {
+
+        if (key.toLowerCase() === paramName.toLowerCase()) {
+            keyName = key;
+
+            return true;
+        }
+    });
+
```

Comment:
Assert that the expected parameter was actually found.

Is this the approach to case-insensitive comparison that is used in other tasks? (a link to another task would be useful)